src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cleanAgentInventory.hpp

```diff
@@ -0,0 +1,123 @@
+     * @param inventoryDatabase Inventory database.
+     * @param subOrchestration Sub-orchestration to publish changes to the indexer.
+     */
+    explicit TCleanAgentInventory(Utils::RocksDBWrapper& inventoryDatabase,
+                                  std::shared_ptr<TAbstractHandler> subOrchestration)
+        : m_subOrchestration(std::move(subOrchestration))
+        , TInventorySync<TScanContext>(inventoryDatabase)
+    {
+    }
+
+    /**
+     * @brief Handles request and passes control to the next step of the chain.
+     *
+     * This method handles cleanup requests based on the operation type specified in the input data.
+     * If the operation type is DeleteAll, it deletes all entries from the inventory database, logs the deletions and
+     * publishes the changes to the indexer. If the operation type is SingleDeleteAndInsert, it performs specific
+     * cleanup actions for entries related to a particular agent, logs those deletions and publishes the changes to the
+     * indexer. If the operation type is not recognized, it throws a std::runtime_error.
+     *
+     * @param data Scan context.
+     * @return std::shared_ptr<TScanContext> Abstract handler.
+     */
+    std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
+    {
+        std::string agentKey;
+        agentKey.append(data->agentNodeName());
+        agentKey.append("_");
+        agentKey.append(data->agentId());
+        agentKey.append("_");
+
+        data->m_isInventoryEmpty = true;
+
+        auto deleteAll = [this](std::shared_ptr<TScanContext> data,
+                                const std::string& key,
+                                const std::vector<AffectedComponentType>& types)
+        {
+            for (const auto& type : types)
+            {
+                for (const auto& dbQuery :
+                     TInventorySync<TScanContext>::m_inventoryDatabase.seek(key, AFFECTED_COMPONENT_COLUMNS.at(type)))
+                {
+                    auto listCve = Utils::split(dbQuery.second.ToString(), ',');
+                    for (const auto& cve : listCve)
+                    {
+                        data->m_isInventoryEmpty = false;
+                        std::string elementKey;
+                        elementKey.append(dbQuery.first);
+                        elementKey.append("_");
+                        elementKey.append(cve);
+
+                        // We use elementKey as the id of the element, because it is unique.
+                        auto context = std::make_shared<TScanContext>();
+                        context->m_elements.emplace(elementKey,
+                                                    TInventorySync<TScanContext>::buildElement("DELETED", elementKey));
+                        m_subOrchestration->handleRequest(std::move(context));
+                    }
+                    logDebug2(WM_VULNSCAN_LOGTAG, "Deleting all agent vulnerabilities key: %s", dbQuery.first.c_str());
```

Comment:
Should this log message be improved and contextualized depending on what type of element was deleted?